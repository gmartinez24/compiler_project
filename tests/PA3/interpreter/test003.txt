// expressions within function call and return for all types

main

int x, y, z;
float a, b, c;
bool i, j, k;

{
    // int
    x = 1;
    y = 2;
    z = 3;

    call printInt(x + y + z);

    // float
    a = 1.0;
    b = 2.1;
    c = 3.2;

    call printFloat(a + b + c);

    // bool
    i = true;
    j = false;
    k = true;

    call printBool(i and j or k);

    // mixed
    call printBool((x + y + z < x - y - z) or i and not(j or k));

    // func call
    call printInt(x + call readInt() + z);

    // return
    return (x + y + z);
}.
