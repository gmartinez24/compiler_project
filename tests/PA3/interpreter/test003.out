Error type-checking file.
TypeError(38,5)[Function main returns int instead of void.]

